servers/zms/src/main/java/com/yahoo/athenz/zms/store/file/FileConnection.java

```diff
@@ -1501,4 +1501,46 @@ public DomainRoleMembers listDomainRoleMembers(String domainName) {
         }
         return domainRoleMembers;
     }
+
+    @Override
+    public boolean confirmRoleMember(String domainName, String roleName, RoleMember member,
+                                    String admin, String auditRef) {
+
+        DomainStruct domainStruct = getDomainStruct(domainName);
+        if (domainStruct == null) {
+            throw ZMSUtils.error(ResourceException.NOT_FOUND, "domain not found", "confirmRoleMember");
+        }
+        Role role = getRoleObject(domainStruct, roleName);
+        if (role == null) {
+            throw ZMSUtils.error(ResourceException.NOT_FOUND, "role not found", "confirmRoleMember");
+        }
+        if (!validatePrincipalDomain(member.getMemberName())) {
```

Comment:
no need for this check here since we're going to look for the inactive member before making any changes.